tools/spm_parser fails building with swiftc 5.7

We just switched to Xcode 14.0.0 and `spm_parser` now fails to build.

```
Building for debugging...
/Users/xdev/gabriel/rules_spm/tools/spm_parser/Sources/spm_parser/Dump.swift:8:8: error: ambiguous use of 'main'
struct Dump: ParsableCommand {
       ^
/Users/xdev/gabriel/rules_spm/tools/spm_parser/Sources/spm_parser/Dump.swift:8:8: error: ambiguous use of 'main'
struct Dump: ParsableCommand {
       ^
[2/3] Compiling spm_parser Dump.swift
```

I'm not exactly sure what could be going on, it might be an issue with a 3rd party dependency, but I feel like you're in a better place to give this a look 🙏🏼

--- a/tools/spm_parser/Sources/spm_parser/AsyncMain.swift
+++ b/tools/spm_parser/Sources/spm_parser/AsyncMain.swift
@@ -1,0 +1,8 @@
+import ArgumentParser
+
+@main
+enum AsyncMain {
+    static func main() async {
+        await Dump.main()
+    }
+}

--- a/tools/spm_parser/Sources/spm_parser/Dump.swift
+++ b/tools/spm_parser/Sources/spm_parser/Dump.swift
@@ -4,7 +4,6 @@ import Foundation
 import TSCBasic
 import Workspace
 
-@main
 struct Dump: AsyncParsableCommand {
     enum DumpError: Error {
         case failedToCreateOutputFile(String)

